react/useAddressRules.tsx

```diff
@@ -1,15 +1,34 @@
 import { useMemo } from 'react'
 import { useQuery } from 'react-apollo'
+import {
+  CountryData,
+  DisplayData,
+  AddressFieldsData,
+  LocationSelectData,
+} from 'vtex.country-data-settings'
 
 import addressRulesQuery from './graphql/addressRulesQuery.gql'
 
+interface RulesQuery {
+  allCountriesData: CountryData[]
+}
+
 const useAddressRules = () => {
-  const { data } = useQuery(addressRulesQuery)
+  const { data } = useQuery<RulesQuery>(addressRulesQuery)
 
   const addressRules = useMemo(
     () =>
-      data?.allCountriesData.reduce(
-        (rules: any, countryData: any) => ({
+      data?.allCountriesData?.reduce<
+        Record<
+          string,
+          {
+            display: DisplayData
+            fields: AddressFieldsData
+            locationSelect?: LocationSelectData | null
+          }
+        >
```

Comment:
Making the typing like this seems to break https://github.com/vtex-apps/checkout-shipping/blob/8e6e8a856c9627c8a7aae96ba049d014f8e3d380/react/ShippingForm.tsx#L181

> **(JSX attribute) AddressContextProps.rules: AddressRules**
> Type 'Record<string, { display: DisplayData; fields: AddressFieldsData; locationSelect?: LocationSelectData | null | undefined; }> | undefined' is not assignable to type 'AddressRules'.
>  Type 'undefined' is not assignable to type 'AddressRules'
> **AddressContext.d.ts(16, 5):** The expected type comes from property 'rules' which is declared here on type 'IntrinsicAttributes & AddressContextProps & { children?: ReactNode; }'

Reply:
I think this is fine, the type for `useAddressRules` inside `checkout-shipping` was allowed to return `undefined`. vtex-apps/checkout-shipping#27 fixes this already

Reply:
I see, that looks good during runtime, but the types for AddressContextProps would still need to be updated, right?

Reply:
I think it should continue to not allow `undefined` as rule values, and we could update the places that we use this hook to add a loading state same as I did on checkout-shipping